app/controllers/concerns/application_controller/authorization.rb

```diff
@@ -43,4 +43,15 @@ def handle_access_denied(exception)
       redirect_to(unauthorized_path)
     end
   end
-end
\ No newline at end of file
+
+  def require_recent_login(options={})
+    unless current_user && current_user.current_login_recent?(options[:max_age])
+      raise RecentLoginRequiredError
+    end
+  end
```

Comment:
I think I mainly did that because it aligned with how Authlogic and Cancan are doing things. Changing it to call `handle_recent_login_required` directly would be fine too.